### PR TITLE
[Feature] RespondWithModal() which accepts an IModal instance as template

### PR DESCRIPTION
**Pull request by Cenngo**
Example usage:
```csharp
[SlashCommand("summon-modal", "desc")]
public async Task SummonModalAsync()
{
    Context.Interaction.RespondWithModal("modal_id", new TestModal
    {
        FirstField = "Initial Value1",
        SecondField = "Initial Value2"
    });
}
```

### CODE REVIEW

File: src/Discord.Net.Interactions/Extensions/IDiscordInteractionExtensions.cs [hunk 1]
```diff
@@ -44,6 +44,44 @@ public static async Task RespondWithModalAsync<T>(this IDiscordInteraction inter
             await SendModalResponseAsync(interaction, customId, modalInfo, options, modifyModal);
         }
 
+        /// <summary>
+        ///     Respond to an interaction with a <see cref="IModal"/> and fills the value fields of the modal using the property values of the provided
```
Comment (csmir) [inline src/Discord.Net.Interactions/Extensions/IDiscordInteractionExtensions.cs]:
```suggestion
        ///     Respond to an interaction with an <see cref="IModal"/> and fills the value fields of the modal using the property values of the provided
```
🤓

File: src/Discord.Net.Interactions/Extensions/IDiscordInteractionExtensions.cs [hunk 2]
```diff
@@ -44,6 +44,44 @@ public static async Task RespondWithModalAsync<T>(this IDiscordInteraction inter
...
+        ///     instance.
+        /// </summary>
+        /// <typeparam name="T">Type of the <see cref="IModal"/> implementation.</typeparam>
+        /// <param name="interaction">The interaction to respond to.</param>
+        /// <param name="modal"><see cref="IModal"/> instance to get field values from.</param>
```
Comment (csmir) [inline src/Discord.Net.Interactions/Extensions/IDiscordInteractionExtensions.cs]:
```suggestion
        /// <param name="modal">The <see cref="IModal"/> instance to get field values from.</param>
```

File: src/Discord.Net.Interactions/Extensions/IDiscordInteractionExtensions.cs [hunk 3]
```diff
@@ -44,6 +44,44 @@ public static async Task RespondWithModalAsync<T>(this IDiscordInteraction inter
...
+        /// <param name="options">The request options for this <see langword="async"/> request.</param>
+        /// <param name="modifyModal">Delegate that can be used to modify the modal.</param>
```
Comment (csmir) [inline src/Discord.Net.Interactions/Extensions/IDiscordInteractionExtensions.cs:55]: Why is `modifyModal` after `options`? Common library behavior places options last.

**Review by csmir — CHANGES_REQUESTED** [review]

File: src/Discord.Net.Interactions/Extensions/IDiscordInteractionExtensions.cs (hunk 3 above)
Comment (Cenngo) [reply src/Discord.Net.Interactions/Extensions/IDiscordInteractionExtensions.cs:55]: factory delegate was an afterthought and was added a good amount of time after the Modals commit so i didnt want it to break existing code bases

Comment (csmir) [reply src/Discord.Net.Interactions/Extensions/IDiscordInteractionExtensions.cs:55]: This should be marked to be changed for v4. I will open an issue. Next to that, lgtm

**Review by csmir — APPROVED** [review]

**Pull request merged**